src/map/tileset.cpp

```diff
@@ -699,20 +718,66 @@ uint32_t CTileset::getQuadFromTile(graphic_index tile) const
 	return base | (base << 8) | (base << 16) | (base << 24);
 }
 
-void CTileset::fillSolidTiles(std::vector<unsigned int> *solidTiles) const
+std::vector<tile_index> CTileset::queryAllTiles() const
 {
-	std::vector<int> seen_types;
-	seen_types.resize(solidTerrainTypes.size(), 0);
-	const size_t tilesCount = getTileCount();
-	for (size_t i = 16; i < tilesCount; i++) {
-		const CTileInfo &info = tiles[i].tileinfo;
-		if (info.BaseTerrain && info.MixTerrain == 0) {
-			if (seen_types[info.BaseTerrain] == 0) {
-				seen_types[info.BaseTerrain] = 1;
-				solidTiles->push_back(tiles[i].tile);
+	std::vector<tile_index> result;
+
+	tile_index tileIdx = 0;
+	for (auto &tile : tiles) {
+		if (tileIdx >= 0x10) { /// First 16 tiles for fog of war.
+			if (tile.tile != 0) { /// Check for separator between tiles
+				result.push_back(static_cast<tile_index>(tileIdx));
```

Comment:
It is already of that type...

Reply:
fixed
https://github.com/Wargus/stratagus/pull/732/commits/d9a4788a898b8cf6b7e6035ed2d0c6667e0489a3